### ISSUE TITLE
"system error: Invalid argument" when attempting to resolve some names without specifying RR type

### ISSUE THREAD

**Issue by lheckemann** [systemd/systemd#32912]
### systemd version the issue has been seen with

254.10

### Used distribution

NixOS 23.11

### Linux kernel version used

6.8.9

### CPU architectures issue was seen on

x86_64

### Component

systemd-resolved

### Expected behaviour you didn't see

Successful resolution of `wifi.sncf`

### Unexpected behaviour you saw

resolved would return errors when attempting to resolve a domain without specifying the RR type, even though it worked fine with an RR type specified and there was no obvious failure visible in resolve's debug logs.

```
[linus@geruest:~]$ resolvectl query wifi.sncf
wifi.sncf: resolve call failed: Lookup failed due to system error: Invalid argument

[linus@geruest:~]$ resolvectl query --type=A wifi.sncf
wifi.sncf IN A 10.32.0.1                                    -- link: wlp170s0

-- Information acquired via protocol DNS in 1.8ms.
-- Data is authenticated: no; Data was acquired via local or encrypted transport: no
-- Data from: cache

[linus@geruest:~]$ resolvectl query --type=AAAA wifi.sncf
wifi.sncf: resolve call failed: Name 'wifi.sncf' does not have any RR of the requested type
```

Other clients such as curl (`curl http://wifi.sncf`) similarly fail to resolve the name.

### Steps to reproduce the problem

Attempt to resolve wifi.sncf while on the wifi of a TGV. What else is necessary to trigger this failure mode is unclear to me.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
varlink: New incoming connection.
varlink: Connections of user 995: 0 (of 576 max)
varlink-18: Setting state idle-server
varlink-18: New incoming message: {"method":"io.systemd.Resolve.ResolveHostname","parameters":{"name":"wifi.sncf","flags":0}}
varlink-18: Changing state idle-server → processing-method
idn2_lookup_u8: wifi.sncf → wifi.sncf
Looking up RR for wifi.sncf IN A.
Looking up RR for wifi.sncf IN AAAA.
Removing cache entry for wifi.sncf IN A (expired 3s ago)
Cache miss for wifi.sncf IN A
Firing regular transaction 56873 for <wifi.sncf IN A> scope dns on wlp170s0/* (validate=yes).
Using feature level UDP+EDNS0 for transaction 56873.
Using DNS server 10.32.0.1 for transaction 56873.
Announcing packet size 1472 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 40
Sending query packet with id 56873 of size 38.
Cache miss for wifi.sncf IN AAAA
Firing regular transaction 4396 for <wifi.sncf IN AAAA> scope dns on wlp170s0/* (validate=yes).
Using feature level UDP+EDNS0 for transaction 4396.
Using DNS server 10.32.0.1 for transaction 4396.
Announcing packet size 1472 in egress EDNS(0) packet.
Emitting UDP, link MTU is 1500, socket MTU is 0, minimal MTU is 40
Sending query packet with id 4396 of size 38.
varlink-18: Changing state processing-method → pending-method
Received dns UDP packet of size 113, ifindex=5, ttl=0, fragsize=0, sender=10.32.0.1, destination=10.32.0.25
Processing incoming packet of size 113 on transaction 4396 (rcode=SUCCESS).
Regular transaction 4396 for <wifi.sncf IN AAAA> on scope dns on wlp170s0/* now complete with <success> from network (unsigned; non-confidential).
Received dns UDP packet of size 129, ifindex=5, ttl=0, fragsize=0, sender=10.32.0.1, destination=10.32.0.25
Processing incoming packet of size 129 on transaction 56873 (rcode=SUCCESS).
Added positive unauthenticated non-confidential cache entry for wifi.sncf IN A 1s on wlp170s0/INET/10.32.0.1
Regular transaction 56873 for <wifi.sncf IN A> on scope dns on wlp170s0/* now complete with <success> from network (unsigned; non-confidential).
Freeing transaction 4396.
varlink-18: Sending message: {"error":"io.systemd.System","parameters":{"errno":22}}
varlink-18: Changing state pending-method → idle-server
Freeing transaction 56873.
varlink-18: Got POLLHUP from socket.
varlink-18: Changing state idle-server → pending-disconnect
varlink-18: Changing state pending-disconnect → processing-disconnect
varlink-18: Changing state processing-disconnect → disconnected
```

**Comment by lheckemann** [comment]
Looks like the relevant code changed between v254 and v255: 8ec951e8d5cdd3ad632b1cbd8bcbe21d68b17512

I might be able to give it a shot, but that depends a little on train WiFi conditions...

**Comment by yuwata** [comment]
Yeah, should be already fixed by 8ec951e8d5cdd3ad632b1cbd8bcbe21d68b17512 (#29307). Closing.
Feel free to request to reopen this when you can reproduce the issue with v255 or newer.
